applications/integrationtests/mechanics/QuasistaticProblem1D.cpp

```diff
@@ -0,0 +1,203 @@
+            EngineeringStrain<1> strain = cellIpData.GetBMatrixStrain(mDof) * cellData.GetNodeValues(mDof);
+            mLaw.Update(strain, dt, cellData.GetCellId(), cellIpData.GetIpId());
+        };
+
+        mEquations.AddGradientFunction(mCellGroup, Gradient);
+        mEquations.AddHessian0Function(mCellGroup, Hessian0);
+        mEquations.AddUpdateFunction(mCellGroup, UpdateHistory);
+
+        auto constraints = DefineConstraints(mMesh, mDof);
+
+        mProblem.SetConstraints(constraints);
+        mProblem.mTolerance = 1.e-12;
+    }
+
+    void SetImperfection(double kappaImperfection)
+    {
+        const int imperfectionCell = mMesh.Elements.Size() / 2;
+        const int imperfectionIp = mLaw.mEvolution.Ip(imperfectionCell, 0);
+        mLaw.mEvolution.mKappas[imperfectionIp] = kappaImperfection;
+    }
+
+    void Solve(double tEnd)
+    {
+        double globalTime = 0;
+        double timeStep = 0.01;
+        double maxTimeStep = 0.1;
+
+        int iStep = 0;
+        while (globalTime < tEnd && timeStep > 1.e-6)
+        {
+            std::cout << "Step " << iStep << " at t = " << globalTime << std::endl;
+
+            auto numIterations = mProblem.DoStep(globalTime + timeStep);
+
+            if (not numIterations)
+            {
+                timeStep *= 0.5;
+                std::cout << "No convergence!\n";
+                continue; // without updating the global time
+            }
+
+            std::cout << "Converence after " << numIterations.get() << " iterations.\n";
+            globalTime += timeStep;
+            iStep++;
+
+            if (numIterations < 3)
+            {
+                timeStep *= 1.5;
+                timeStep = std::min(timeStep, maxTimeStep);
+                std::cout << "--> Increasing time step to " << timeStep << ".\n";
+            }
+        }
+    }
+
+    std::vector<double> DamageField()
+    {
+        auto Damage = [&](const CellData& cellData, const CellIpData& cellIpData) {
+            int ip = mLaw.mEvolution.Ip(cellData.GetCellId(), cellIpData.GetIpId());
+            double kappa = mLaw.mEvolution.mKappas[ip];
+            return Eigen::VectorXd::Constant(1, mLaw.mDamageLaw.Damage(kappa));
+        };
+
+        std::vector<double> damage;
+        for (auto& cell : mCellGroup)
+        {
+            auto ipDamage = cell.Eval(Damage);
+            for (auto dmg : ipDamage)
+                damage.push_back(dmg[0]);
+        }
+        return damage;
+    }
+
+private:
+    using ElasticLaw = Laws::LinearElastic<1>;
+    using DamageLaw = Constitutive::DamageLawExponential;
+    using StrainNorm = Constitutive::ModifiedMisesStrainNorm<1>;
+    using Evolution = Laws::EvolutionImplicit<1, StrainNorm>;
+    using Law = Laws::LocalIsotropicDamage<1, DamageLaw, Evolution, ElasticLaw>;
+
+    MeshFem mMesh;
+
+    DofType mDof;
+    Law mLaw;
+    Integrands::MomentumBalance<1> mMomentumBalance;
+
+    TimeDependentProblem mEquations;
+    QuasistaticSolver mProblem;
```

Comment:
This naming here is confusing the class TimeDependent problem is mEquations, and the Solver is called problem

Reply:
That was misses during renaming, I guess. However, I am not happy with names `TimeDependentProblem` and `QuasistaticSolver` either. :cry: 
I expect a huge redesign when we tackle #141, where we can rethink those names.